### PR TITLE
enforce variable naming style on everything I could find

### PR DESCRIPTION
**Pull request by LeftofZen**

### CODE REVIEW

File: src/Thirdparty/filesystem.hpp [hunk 1]
```diff
@@ -207,7 +207,7 @@
 #include <utility>
 #include <vector>
 
-#else  // GHC_EXPAND_IMPL
+#else // GHC_EXPAND_IMPL
```
Comment (LeftofZen) [inline src/Thirdparty/filesystem.hpp:210]: oops, revert this file

File: src/OpenLoco/GameCommands/RemoveBuilding.cpp [hunk 2]
```diff
@@ -75,7 +75,7 @@ namespace OpenLoco::GameCommands
                     }
                 }
             }
-            auto animOffsets = getUnk4F9274(buildingObj->flags & BuildingObjectFlags::large_tile);
+            auto animOffsets = getUnk4F9274(buildingObj->flags & BuildingObjectFlags::largeTle);
```
Comment (LeftofZen) [inline src/OpenLoco/GameCommands/RemoveBuilding.cpp:78]: typo on largeTile

File: src/OpenLoco/Graphics/Colour.h [hunk 3]
```diff
@@ -195,9 +195,9 @@ namespace OpenLoco
 
     struct AdvancedColour
     {
-        static constexpr uint8_t outline_flag = 1 << 5;
-        static constexpr uint8_t inset_flag = 1 << 6;
-        static constexpr uint8_t translucent_flag = 1 << 7;
+        static constexpr uint8_t kOutlineFlag = 1 << 5;
```
Comment (LeftofZen) [inline src/OpenLoco/Graphics/Colour.h:198]: undo k prefix for flags

File: src/OpenLoco/Localisation/StringIds.h [hunk 4]
```diff
@@ -422,7 +422,7 @@ namespace OpenLoco::StringIds
     constexpr string_id quantity_seven_eigths = 438;
     constexpr string_id comma = 439;
     constexpr string_id produces = 440;
-    constexpr string_id under_construction = 441;
+    constexpr string_id underConstruction = 441;
```
Comment (LeftofZen) [inline src/OpenLoco/Localisation/StringIds.h:425]: somehow I've accidentally renamed these string_ids, will revert

File: src/OpenLoco/Objects/TrainStationObject.h [hunk 5]
```diff
@@ -59,8 +59,8 @@ namespace OpenLoco
 
     namespace TrainStation::ImageIds
     {
-        constexpr uint32_t preview_image = 0;
-        constexpr uint32_t preview_image_windows = 1;
+        constexpr uint32_t previewImage = 0;
```
Comment (LeftofZen) [inline src/OpenLoco/Objects/TrainStationObject.h:62]: make static and k-prefix

File: src/OpenLoco/Scenario.h [hunk 6]
```diff
@@ -44,63 +44,63 @@ namespace OpenLoco::Scenario
...
+    constexpr uint32_t kMaxObjectivePerformanceIndex = 100;
 
-    constexpr uint32_t min_objective_delivered_cargo = 100;
-    constexpr uint32_t max_objective_delivered_cargo = 200000000;
+    constexpr uint32_t kMinObjectiveDeliveredCargo = 100;
+    constexpr uint32_t kMaxObjectiveDeliveredCargo = 200000000;
 
-    constexpr uint8_t min_competing_companies = 0;
+    constexpr uint8_t kMinCompetingCompanies = 0;
     constexpr uint8_t max_competing_companies = 14;
```
Comment (LeftofZen) [inline src/OpenLoco/Scenario.h:67]: wrong style, as rename got confused between these variabels and the string_ids

File: src/OpenLoco/ZoomLevel.hpp [hunk 7]
```diff
@@ -5,6 +5,15 @@
 
 namespace OpenLoco
 {
+    /*enum class ZoomLevel : uint8_t
```
Comment (LeftofZen) [inline src/OpenLoco/ZoomLevel.hpp:8]: remove commented code

File: src/OpenLoco/Audio/Audio.cpp [hunk 8]
```diff
@@ -99,36 +99,36 @@ namespace OpenLoco::Audio
...
+        { PathId::music_a_good_head_of_steam, StringIds::music_a_good_head_of_steam, 1950, 1958 },
+        { PathId::music_hop_to_the_bop, StringIds::music_hop_to_the_bop, 1946, 1954 },
+        { PathId::music_the_city_lights, StringIds::music_the_city_lights, 1980, 1988 },
+        { PathId::music_steamin_down_town, StringIds::music_steamin_down_town, 1960, 1968 },
+        { PathId::music_bright_expectations, StringIds::music_bright_expectations, 1983, 1991 },
+        { PathId::music_mo_station, StringIds::music_mo_station, 1963, 1971 },
+        { PathId::music_far_out, StringIds::music_far_out, 1966, 1974 },
+        { PathId::music_running_on_time, StringIds::music_running_on_time, 1986, 1994 },
+        { PathId::music_get_me_to_gladstone_bay, StringIds::music_get_me_to_gladstone_bay, 1918, 1926 },
+        { PathId::music_sandy_track_blues, StringIds::music_get_me_to_gladstone_bay, 1921, 1929 }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Audio/Audio.cpp:131]:
```suggestion
        { PathId::music_sandy_track_blues, StringIds::music_sandy_track_blues, 1921, 1929 }
```

File: src/OpenLoco/Localisation/StringManager.h [hunk 9]
```diff
@@ -31,38 +31,38 @@ namespace OpenLoco
...
-        constexpr uint8_t font_large = 8;
-        constexpr uint8_t font_bold = 9;
-        constexpr uint8_t font_regular = 10;
+        constexpr uint8_t newlineSmaller = 6;
+        constexpr uint8_t fontSmall = 7;
+        constexpr uint8_t fontLarge = 8;
+        constexpr uint8_t fontBold = 9;
+        constexpr uint8_t fontRegular = 10;
         constexpr uint8_t outline = 11;
-        constexpr uint8_t outline_off = 12;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Localisation/StringManager.h:48]: Maybe make a Font namespace for these as well?

Comment (LeftofZen) [reply src/OpenLoco/Localisation/StringManager.h:48]: I thought about it haha but in the end didn't. Happy to do that when I get home.

Comment (LeftofZen) [reply src/OpenLoco/Localisation/StringManager.h]: done

File: src/OpenLoco/Objects/TrainStationObject.h [hunk 10]
```diff
@@ -59,8 +59,8 @@ namespace OpenLoco
 
     namespace TrainStation::ImageIds
     {
-        constexpr uint32_t preview_image = 0;
-        constexpr uint32_t preview_image_windows = 1;
+        static constexpr uint32_t kPreviewImage = 0;
+        static constexpr uint32_t kPreviewImageWindows = 1;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Objects/TrainStationObject.h:63]: Why were these made static?

File: src/OpenLoco/Utility/String.hpp [hunk 11]
```diff
@@ -55,53 +55,53 @@ namespace OpenLoco::Utility
...
+            return srcLen;
         }
 
         // this lambda is essentially a reimplementation of strnlen, which isn't standard
-        size_t dest_len = [=] {
-            auto dest_end = reinterpret_cast<const char*>(std::memchr(dest, '\0', size));
-            if (dest_end != nullptr)
+        size_t destLen = [=] {
+            auto destLen = reinterpret_cast<const char*>(std::memchr(dest, '\0', size));
+            if (destLen != nullptr)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Utility/String.hpp:85]: `dest_end` has been changed into `destLen`, which is the same name as the outer variable. Mistake?

File: src/OpenLoco/Window.h [hunk 12]
```diff
@@ -267,7 +267,7 @@ namespace OpenLoco::Ui
         int16_t rowInfo[1000];                             // 0x6A
         uint16_t rowCount;                                 // 0x83A
         uint16_t var_83C;
-        uint16_t rowHeight;    // 0x83E
+        uint16_t kRowHeight;   // 0x83E
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Window.h:270]: Accidental rename?

Comment (LeftofZen) [reply src/OpenLoco/Window.h:270]: yep, looks like VS renamed this one as well as the other variables with the same name which were constants

File: src/OpenLoco/Utility/String.hpp (hunk 11 above)
Comment (LeftofZen) [reply src/OpenLoco/Utility/String.hpp:85]: it is a mistake but interestingly due to the lambda scope, it's a separate variable. fixed now

File: src/OpenLoco/Objects/TrainStationObject.h (hunk 10 above)
Comment (LeftofZen) [reply src/OpenLoco/Objects/TrainStationObject.h:63]: code style - everything else being made constexpr is also static. should I not make it static?

**Review by AaronVanGeffen — APPROVED** [review]
Thanks for tackling these. I've gone over the changes and believe them to be in order. However, given the scope, I think someone else should go over them as well.

**Comment by LeftofZen** [comment]
Yeah I was in two minds about the scope, on one hand its obviously out of control, but on the other hand I don't really wanna make another 10 'code style' PRs

**Comment by LeftofZen** [comment]
I will also go over every single change I made again tonight, before I merge

File: src/OpenLoco/Audio/Audio.cpp [hunk 13]
```diff
@@ -99,36 +99,36 @@ namespace OpenLoco::Audio
...
-        { PathId::music_80s3, StringIds::music_bright_expectations, 1983, 1991 },
-        { PathId::music_60s2, StringIds::music_mo_station, 1963, 1971 },
-        { PathId::music_60s3, StringIds::music_far_out, 1966, 1974 },
-        { PathId::music_80s4, StringIds::music_running_on_time, 1986, 1994 },
-        { PathId::music_20s5, StringIds::music_get_me_to_gladstone_bay, 1918, 1926 },
-        { PathId::music_20s6, StringIds::music_sandy_track_blues, 1921, 1929 }
+        { PathId::music_the_ragtime_dance, StringIds::music_the_ragtime_dance, 1909, 1917 },
+        { PathId::music_easy_winners, StringIds::music_easy_winners, 0, 1914 },
+        { PathId::music_setting_off, StringIds::music_setting_off, 1929, 1937 },
+        { PathId::music_a_travellers_seranade, StringIds::music_a_travellers_seranade, 1940, 1948 },
```
Comment (LeftofZen) [inline src/OpenLoco/Audio/Audio.cpp:120]: serenade is spelt wrong here

File: src/OpenLoco/Objects/DockObject.h [hunk 14]
```diff
@@ -29,8 +29,8 @@ namespace OpenLoco
         uint8_t num_aux_02_ent; // 0x13
```
Comment (LeftofZen) [inline src/OpenLoco/Objects/DockObject.h:29]: looks like I missed these 2 in the rename, but they become somewhat ugly - `numAux01` and `numAux02Ent`?

File: src/OpenLoco/Objects/TrainStationObject.h (hunk 10 above)
Comment (LeftofZen) [reply src/OpenLoco/Objects/TrainStationObject.h:63]: I found other ImageIds that weren't static, so I think I'll revert the static here

File: src/OpenLoco/Ui/WindowType.h [hunk 15]
```diff
@@ -60,7 +60,7 @@ namespace OpenLoco::Ui
         confirmDisplayModePrompt = 50,
         textInput = 51,
         fileBrowserPrompt = 52,
-        previewImage = 53,
+        kPreviewImage = 53,
```
Comment (LeftofZen) [inline src/OpenLoco/Ui/WindowType.h:63]: this one probably doesn't need the k prefix

File: src/OpenLoco/Window.cpp [hunk 16]
```diff
@@ -619,36 +619,36 @@ namespace OpenLoco::Ui
 
         // Apply offset to the viewport.
         ViewportConfig* vc = &this->viewportConfigurations[0];
-        vc->saved_view_x = dest.x + rebased_x + (offset_x / (1 << v->zoom));
```
Comment (LeftofZen) [inline src/OpenLoco/Window.cpp:622]: missed offset_x and offset_y

Comment (LeftofZen) [reply src/OpenLoco/Window.cpp:622]: missed a lot in this file actually

File: src/OpenLoco/Window.h [hunk 17]
```diff
@@ -412,26 +412,26 @@ namespace OpenLoco::Ui
         void callClose();                                                                              // 0
         void callOnMouseUp(WidgetIndex_t widgetIndex);                                                 // 1
         Ui::Window* callOnResize();                                                                    // 2
-        void call_3(int8_t widget_index);                                                              // 3
-        void callOnMouseDown(int8_t widget_index);                                                     // 4
-        void callOnDropdown(WidgetIndex_t widget_index, int16_t item_index);                           // 5
+        void call_3(int8_t widgetIndex);                                                               // 3
+        void callOnMouseDown(int8_t widgetIndex);                                                      // 4
+        void callOnDropdown(WidgetIndex_t widgetIndex, int16_t item_index);                            // 5
```
Comment (LeftofZen) [inline src/OpenLoco/Window.h:417]: missed item_index

File: src/OpenLoco/Windows/Options.cpp [hunk 18]
```diff
@@ -579,15 +579,15 @@ namespace OpenLoco::Ui::Windows::Options
             else
                 w.widgets[Widx::construction_marker].text = StringIds::white;
 
-            static const string_id scale_string_ids[] = {
+            static const string_id scaleStringIds[] = {
```
Comment (LeftofZen) [inline src/OpenLoco/Windows/Options.cpp:582]: can be constexpr

File: src/OpenLoco/Windows/TerraForm.cpp [hunk 19]
```diff
@@ -1885,7 +1885,7 @@ namespace OpenLoco::Ui::Windows::Terraform
     {
         static constexpr Ui::Size kWindowSize = { 418, 108 };
 
-        static const uint8_t rowHeight = 48;
+        static const uint8_t kRowHeight = 48;
```
Comment (LeftofZen) [inline src/OpenLoco/Windows/TerraForm.cpp:1888]: constexpr

File: src/OpenLoco/Windows/TownList.cpp [hunk 20]
```diff
@@ -74,10 +74,10 @@ namespace OpenLoco::Ui::Windows::TownList
     namespace TownList
     {
         static constexpr Ui::Size kWindowSize = { 600, 197 };
-        static constexpr Ui::Size maxDimensions = { 600, 900 };
-        static constexpr Ui::Size minDimensions = { 192, 100 };
+        static constexpr Ui::Size kMaxDimensions = { 600, 900 };
+        static constexpr Ui::Size kMinDimensions = { 192, 100 };
 
-        static const uint8_t rowHeight = 10;
+        static const uint8_t kRowHeight = 10;
```
Comment (LeftofZen) [inline src/OpenLoco/Windows/TownList.cpp:80]: constexpr

**Comment by LeftofZen** [comment]
I've corrected a bunch of small mistakes and missed renames, I think this is as good as its gonna get

File: src/OpenLoco/Paint/Paint.h [hunk 21]
```diff
@@ -200,8 +200,8 @@ namespace OpenLoco::Paint
         /*
          * @param rotation @<ebp>
          * @param imageId  @<ebx>
-         * @param offset_x @<al>
-         * @param offset_y @<cl>
+         * @param offsetX @<al>
+         * @param offsetY @<cl>
          * @param offset_z @<dx>
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Paint/Paint.h:205]: offsetZ

**Comment by AaronVanGeffen** [comment]
> I've corrected a bunch of small mistakes and missed renames/constexprs, I think this is as good as its gonna get

There are a few more areas that require code style changes (e.g. widx enums), but I would suggest to stop adding more to this PR at this point.

I'll review your last set of changes tomorrow.

File: src/OpenLoco/Paint/Paint.h (hunk 21 above)
Comment (LeftofZen) [reply src/OpenLoco/Paint/Paint.h:205]: should we delete these doxygen comments? doesn't seem like they're used or updated

**Comment by LeftofZen** [comment]
> There are a few more areas that require code style changes (e.g. widx enums), but I would suggest to stop adding more to this PR at this point.

Yep I can take a look at them later. Is the coding style page up-to-date? https://github.com/OpenLoco/OpenLoco/wiki/Coding-Style. Or are there any more things to add to it? Just so I can use that as a reference for any future style changes

**Comment by duncanspumpkin** [comment]
> > There are a few more areas that require code style changes (e.g. widx enums), but I would suggest to stop adding more to this PR at this point.
> 
> Yep I can take a look at them later. Is the coding style page up-to-date? https://github.com/OpenLoco/OpenLoco/wiki/Coding-Style. Or are there any more things to add to it? Just so I can use that as a reference for any future style changes

If it's missing something then add to it. Currently there are two main things that don't follow that: interop uses snake case; imageids and stringids use snake case because they read a bit nicer;

File: src/OpenLoco/Localisation/Conversion.cpp [hunk 22]
```diff
@@ -77,13 +77,13 @@ namespace OpenLoco::Localisation
             return '?';
     }
 
-    std::string convertUnicodeToLoco(std::string unicode_string)
+    std::string convertUnicodeToLoco(std::string unicodeString)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Localisation/Conversion.cpp:80]: This should really be passed as a `const &`

File: src/OpenLoco/Objects/InterfaceSkinObject.h [hunk 23]
```diff
@@ -48,7 +48,7 @@ namespace OpenLoco
 
     namespace InterfaceSkin::ImageIds
     {
-        constexpr uint32_t preview_image = 0;
+        constexpr uint32_t kPreviewImage = 0;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Objects/InterfaceSkinObject.h:51]: i think we should keep ImageIds with the snake_case.

File: src/OpenLoco/Objects/LandObject.h [hunk 24]
```diff
@@ -52,7 +52,7 @@ namespace OpenLoco
 
     namespace Land::ImageIds
     {
-        constexpr uint32_t landscape_generator_tile_icon = 1;
-        constexpr uint32_t toolbar_terraform_land = 3;
+        constexpr uint32_t kLandscapeGeneratorTileIcon = 1;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Objects/LandObject.h:55]: same here

File: src/OpenLoco/Objects/RoadStationObject.h [hunk 25]
```diff
@@ -57,7 +57,7 @@ namespace OpenLoco
 
     namespace RoadStation::ImageIds
     {
-        constexpr uint32_t preview_image = 0;
-        constexpr uint32_t preview_image_windows = 1;
+        constexpr uint32_t kPreviewImage = 0;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Objects/RoadStationObject.h:60]: and here

File: src/OpenLoco/Objects/TrainStationObject.h [hunk 26]
```diff
@@ -59,8 +59,8 @@ namespace OpenLoco
 
     namespace TrainStation::ImageIds
     {
-        constexpr uint32_t preview_image = 0;
-        constexpr uint32_t preview_image_windows = 1;
+        constexpr uint32_t kPreviewImage = 0;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Objects/TrainStationObject.h:62]: and here.

**Comment by duncanspumpkin** [comment]
All looks good. I can't quite decide what to do with imageids but we should make them consistent either all snake_case or all camelCase also need to think if they should have a k or if they are really just part of an enum. Not sure on them so I think I'll just approve as is for now.

**Review by duncanspumpkin — APPROVED** [review]

**Comment by LeftofZen** [comment]
Agreed with all your comments and I've made the fixes you pointed out, I'll keep ImageIds as snake_case for now until we decide what we want to do with them.

**Pull request merged**